gridcomps/History/MAPL_HistoryTrajectoryMod.F90

```diff
@@ -6,44 +6,58 @@ module HistoryTrajectoryMod
   use MAPL_VerticalDataMod
   use LocStreamFactoryMod
   use MAPL_LocstreamRegridderMod
-  use, intrinsic :: iso_fortran_env, only: REAL32
   use, intrinsic :: iso_fortran_env, only: REAL64
   implicit none
   private
 
-  public :: HistoryTrajectory
+  public :: obs_unit
+  type :: obs_unit
+     integer :: nobs_epoch
+     type(FileMetadata), allocatable            :: metadata
+     type(NetCDF4_FileFormatter), allocatable   :: file_handle
+     character(len=ESMF_MAXSTR)                 :: name
+     character(len=ESMF_MAXSTR)                 :: obsFile_output
+     character(len=ESMF_MAXSTR)                 :: input_template
+     real(kind=REAL64), allocatable :: lons(:)
+     real(kind=REAL64), allocatable :: lats(:)
+     real(kind=REAL64), allocatable :: times_R8(:)
+     real(kind=REAL32), allocatable :: p2d(:)
+     real(kind=REAL32), allocatable :: p3d(:,:)
+  end type obs_unit
```

Comment:
This can be addressed in a later PR, but to my eyes this derived type is large and complex enough to be worth its own module for treatment.

Reply:
After adding this derived type in, I had two types of thinking: i) the code structure and logic become easier to follow and better organized, ii) it is created mainly because there appears not to be an easy implementation to purge as objects the FileMetadata and NetCDF4_FileFormatter derived type; and the reason to purge them is because the Epoch concept to regenerate metadata and files.  So, the easy way is to allocate and deallocate them within the object.  On the other hand, it is one step towards the obs treatment in the OO way.